salesforce_functions/data_api/_requests.py

```diff
@@ -229,38 +229,45 @@ async def _process_records_response(
         raise SalesforceRestApiError(_parse_errors(json_body))
 
     if isinstance(json_body, dict):
-        done: bool = json_body["done"]
-        total_size: int = json_body["totalSize"]
-        next_records_url: str | None = json_body.get("nextRecordsUrl")
-
-        records: list[QueriedRecord] = []
-        for record_json in json_body["records"]:
-            salesforce_object_type = record_json["attributes"]["type"]
-
-            fields = {}
-            sub_query_results = {}
-            for key, value in record_json.items():
-                if key == "attributes":
-                    continue
-
-                if isinstance(value, dict):
-                    sub_query_results[key] = await _process_records_response(
-                        status_code, cast(dict[str, Any], value), download_file_fn
-                    )
-                elif _is_binary_field(salesforce_object_type, key):
-                    fields[key] = await download_file_fn(value)
-                else:
-                    fields[key] = value
-
-            records.append(
-                QueriedRecord(salesforce_object_type, fields, sub_query_results)
-            )
-
-        return RecordQueryResult(done, total_size, records, next_records_url)
+        return await _parse_record_query_result(json_body, download_file_fn)
 
     raise UnexpectedRestApiResponsePayload()
 
 
+async def _parse_record_query_result(json_body: dict, download_file_fn: DownloadFileFunction) -> RecordQueryResult:
```

Comment:
Mypy is failing with:

```
salesforce_functions/data_api/_requests.py:237: error: Missing type parameters
for generic type "dict"  [type-arg]
    async def _parse_record_query_result(json_body: dict, download_file_fn...
                                                    ^
salesforce_functions/data_api/_requests.py:249: error: Missing type parameters
for generic type "dict"  [type-arg]
    async def _parse_queried_record(record_json: dict, download_file_fn: D...
                                                 ^
salesforce_functions/data_api/_requests.py:264: error: Incompatible types in
assignment (expression has type "bytes", target has type "QueriedRecord") 
[assignment]
                fields[key] = await download_file_fn(value)
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
i figured out how to run `pylint`.  didn't realize `mypy` was another check